mayastor/src/lvs/pool.rs

```diff
@@ -0,0 +1,511 @@
+                // We used to clear a pool with UNMAP but that takes awfully
+                // long time on large SSDs (~ can take an hour). Clearing the
+                // pool is not necessary. Clearing the lvol must be done, but
+                // lvols tend to be small so there the overhead is acceptable.
+                LVS_CLEAR_WITH_NONE,
+                Some(Self::lvs_cb),
+                cb_arg(sender),
+            )
+        }
+        .to_result(|e| Error::Create {
+            err: Errno::from_i32(e),
+            msg: format!("failed to create pool '{}'", name),
+        })?;
+
+        receiver
+            .await
+            .expect("Cancellation is not supported")
+            .to_result(|e| Error::Create {
+                err: Errno::from_i32(e),
+                msg: "failed to create pool".into(),
+            })?;
+
+        match Self::lookup(&name) {
+            Some(pool) => {
+                info!("The pool '{}' has been created on {}", name, bdev);
+                Ok(pool)
+            }
+            None => Err(Error::Create {
+                err: Errno::ENOENT,
+                msg: format!("The pool {} is gone right after creation!", name),
+            }),
+        }
+    }
+
+    /// imports the pool if it exists, otherwise try to create it
+    #[instrument(level = "debug", err)]
+    pub async fn create_or_import(
+        args: CreatePoolRequest,
+    ) -> Result<Lvs, Error> {
+        if args.disks.len() != 1 {
+            return Err(Error::BadNumDisks {
+                num: args.disks.len(),
+            });
+        }
+
+        let parsed =
+            Uri::parse(&args.disks[0]).map_err(|e| Error::InvalidBdev {
+                source: e,
+                msg: "invalid bdev specified for pool".to_string(),
+            })?;
+
+        if let Some(pool) = Self::lookup(&args.name) {
+            return if pool.base_bdev().name() == parsed.get_name() {
+                Ok(pool)
+            } else {
+                Err(Error::Create {
+                    err: Errno::EEXIST,
+                    msg: format!("pool {} already exists", args.name),
+                })
+            };
+        }
+
+        let bdev = match parsed.create().await {
+            Err(e) => match e {
+                NexusBdevError::BdevExists {
+                    ..
+                } => Ok(parsed.get_name()),
+                _ => Err(Error::InvalidBdev {
+                    source: e,
+                    msg: "base_bdev for the pool does not match".into(),
+                }),
+            },
+            Ok(name) => Ok(name),
+        }?;
+
+        match Self::import(&args.name, &bdev).await {
+            Ok(pool) => Ok(pool),
+            Err(_) => Self::create(&args.name, &bdev).await,
+        }
+    }
+
+    /// export the given lvl
+    #[allow(clippy::unit_arg)] // here to silence the () argument
+    #[instrument(level = "debug", err)]
+    pub async fn export(self) -> Result<(), Error> {
```

Comment:
I'm confused about this export... seems more like we're shutting down the pool?

Reply:
Hmm, sorry @tiagolobocastro I just saw this comment while I was closing the pool request... Sorry about that.

export means, unload the pool, destroy the base_bdev -- but leave the data on disk intact such that it can be re-imported.

destroy means the same as above but you cant import it anymore